packages/medusa/src/services/event-bus.ts

```diff
@@ -258,18 +284,22 @@ export default class EventBusService {
         this.stagedJobRepository_
       )
 
-      const jobToCreate = {
-        event_name: eventName,
-        data: data as unknown as Record<string, unknown>,
-        options: opts,
-      } as Partial<StagedJob>
+      const jobToCreates = events.map((event) => {
+        return stagedJobRepository.create({
+          event_name: event.eventName,
+          data: event.data,
+          options: event.options,
+        } as DeepPartial<StagedJob>) as QueryDeepPartialEntity<StagedJob>
+      })
 
-      const stagedJobInstance = stagedJobRepository.create(jobToCreate)
+      const stagedJobs = await stagedJobRepository.insertBulk(jobToCreates)
 
-      return await stagedJobRepository.save(stagedJobInstance)
+      return (isString(eventNameOrData)
+        ? stagedJobs[0]
+        : stagedJobs) as unknown as TResult
     }
 
-    this.queue_.add({ eventName, data }, opts)
+    this.queue_.addBulk(events)
```

Comment:
todo: An `await` has been missing the whole time
```suggestion
    await this.queue_.addBulk(events)
```

Reply:
yes, you are right, just checked the impl